### PR TITLE
Add drop_invalid_header_fields to AWS ALB Resource

### LINKED ISSUE #10862: aws_lb needs to support setting the drop_invalid_header_fields setting
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Based on a change in default behavior to the AWS load balancers described at https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html, the routing.http.drop_invalid_header_fields.enabled defaults to true for load balancers starting on November 11, 2019. 

Due to the AWS implementation, headers that should be valid such as "access_token" are being dropped. This is discussed on the official forums at https://forums.aws.amazon.com/thread.jspa?messageID=923165&#923165

In order to allow load balancers to be created in the previous behavior, the aws_lb resource needs to be able to set the drop_invalid_header_fields setting.

### New or Affected Resource(s)
* aws_lb

### References
* https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html
* https://forums.aws.amazon.com/thread.jspa?messageID=923165&#923165

### PR DESCRIPTION
**Pull request by kireledan**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #10862 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* resource/aws_lb: Add drop_invalid_header_fields argument (#10862)
```

Output from acceptance testing:

```
 ❯ make testacc TEST=./aws TESTARGS='-run=TestAccAWSLB_applicationLoadBalancer_updateDropInvalidHeaderFields'                  [12:11:14]
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLB_applicationLoadBalancer_updateDropInvalidHeaderFields -timeout 120m
=== RUN   TestAccAWSLB_applicationLoadBalancer_updateDropInvalidHeaderFields
=== PAUSE TestAccAWSLB_applicationLoadBalancer_updateDropInvalidHeaderFields
=== CONT  TestAccAWSLB_applicationLoadBalancer_updateDropInvalidHeaderFields
--- PASS: TestAccAWSLB_applicationLoadBalancer_updateDropInvalidHeaderFields (435.52s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       436.321s..
```

### CODE REVIEW

**Comment by jonDowdle** [comment]
@bflad Any chance you could take a look at this?

**Comment by kireledan** [comment]
> Rebased against master

**Comment by jonDowdle** [comment]
Can this get eyes on it? We haven't heard anything back yet.

**Comment by jonDowdle** [comment]
@paultyng I saw you removed the needs-triage label. Is there anything we need to do before this can be merged?

**Comment by BRMatt** [comment]
We came across this today after seeing [a slack security bounty on http request smugging](https://hackerone.com/reports/737140), and a comment about the implications on AWS ALBs [on hackernews](https://news.ycombinator.com/item?id=22568309). Apparently the AWS ALB does not drop invalid headers by default, so it would be good if we could configure this option via terraform.

**Comment by jonDowdle** [comment]
Agreed. This has been waiting for 3 months now. I'm not sure who or what needs to happen for it to get merged.

File: aws/data_source_aws_lb.go [hunk 1]
```diff
@@ -103,6 +103,11 @@ func dataSourceAwsLb() *schema.Resource {
 				Computed: true,
 			},
 
+			"drop_invalid_header_fields": {
```
[the remark this answers is not in the document]
Comment (bflad) [reply aws/data_source_aws_lb.go]: Ah looks like that data source documentation is still using the older style of referencing the resource documentation:

```
See the [LB Resource](/docs/providers/aws/r/lb.html) for details on the
returned attributes - they are identical.
```

So this is fine. 👍

**Pull request merged**